### DISCUSSION TITLE
Speeding up compilation

### DISCUSSION THREAD

**Discussion by Le0X8** [slint-ui/slint#7215] [Q&A]
Hi!

I recently started using Slint, but I noticed that compilation in debug and especially release mode takes up a very long time, even if only one component is being used.

Is there any way to speed up the compilation process like caching builds like Cargo does?

Or are there any recommendations to consider when structuring a Slint project? Putting everything in one file and using the Slint macro everywhere is clearly no option at as a growing codebase with everything in one file gets really messy after the third component.

My `build.rs`:

```rs
use std::{collections::HashMap, path::PathBuf};

fn main() {
    let aliases = HashMap::from(
        [("images", "assets/images"), ("fonts", "assets/fonts")]
            .map(|(k, v)| (k.to_string(), PathBuf::from(v))),
    );

    let cfg = slint_build::CompilerConfiguration::new().with_library_paths(aliases);

    slint_build::compile_with_config("ui/app-window.slint", cfg).expect("Slint build failed");
}
```

My `src/main.rs`:

```rs
#![cfg_attr(not(debug_assertions), windows_subsystem = "windows")]

use std::error::Error;

use slint::set_xdg_app_id;

slint::include_modules!();

fn main() -> Result<(), Box<dyn Error>> {
    let ui = AppWindow::new()?;

    set_xdg_app_id("dev.leox.slint-test")?;

    ui.on_request_increase_value({
        let ui_handle = ui.as_weak();
        move || {
            let ui = ui_handle.unwrap();
            ui.set_counter(ui.get_counter() + 1);
        }
    });

    ui.run()?;

    Ok(())
}
```

**Comment by ogoffart** [comment]
We are aware of the issue that the compilation can be slow. There is a tracking issue: https://github.com/slint-ui/slint/issues/4082
Unfortunately, right now, we don't have a solution.
